chalicelib/es_connection.py

```diff
@@ -0,0 +1,279 @@
+from .abstract_connection import AbstractConnection
+from elasticsearch import (
+    Elasticsearch,
+    TransportError,
+    RequestError,
+    ConnectionTimeout
+    )
+from elasticsearch_dsl import Search
+from dcicutils import es_utils
+from .utils import load_json
+import json
+import time
+
+# configure ES info from here
+HOST = 'https://search-foursight-fourfront-ylxn33a5qytswm63z52uytgkm4.us-east-1.es.amazonaws.com'
+SEARCH_SIZE = 10000
+
+class ElasticsearchException(Exception):
+    """ Generic exception for an elasticsearch failure """
+    def __init__(self, message=None):
+        if message is None:
+            self.message = "No error message given, this shouldn't happen!"
+        else:
+            self.message = message
+        super().__init__(message)
+
+class ESConnection(AbstractConnection):
+    """
+    ESConnection is a handle to a remote ElasticSearch instance on AWS.
+    All Foursight connections make use of the same ES instance but have separate
+    indices for each one, such as 'foursight-dev-cgap', 'foursight-dev-data' etc
+
+    ESConnection is intended to work with only a single index.
+
+    Implements the AbstractConnection 'interface'
+    """
+    def __init__(self, index=None, doc_type='result'):
+        self.es = es_utils.create_es_client(HOST, use_aws_url=True)
+        self.index = index
+        self.mapping = {} # only needed if we create the index
```

Comment:
If `self.mapping` is a property that is only used in `self.create_index`, can we just make it local to that method?